### ISSUE TITLE
Check origin errors are always logged

### ISSUE THREAD

**Issue by thiamsantos** [phoenixframework/phoenix#3937]
### Environment

* Elixir version (elixir -v):

```
Erlang/OTP 22 [erts-10.7.1] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe]

Elixir 1.10.1 (compiled with Erlang/OTP 22)
```

* Phoenix version (mix deps): 

```
* phoenix 1.5.0 (Hex package) (mix)
  locked at 1.5.0 (phoenix) c695e372
  ok
```

* Operating system:

```
Linux Mint 19.2 Tina x86_64
```

### Expected behavior

I find a little strange to always show a complete log to show how to fix the issue, even in production. Seems to me that this is not always a actual issue, the app could be a target of some source of attack, from a domain that is not allowed to access the socket. In this cases the log can be misleading or confusing as it will appear that we are experiencing an configuration error, but in the end the behavior is correct and that connection attempt should be blocked.

I think for local development this logs are really nice, but I'm not sure if they should also appear in production. Maybe a different message could be used in production.

### Actual behavior

When a domain that is not allowed tries to connect to a socket, we get a complete error log explaining how to "fix the issue".

https://github.com/phoenixframework/phoenix/blob/v1.5.4/lib/phoenix/socket/transport.ex#L360

```
[error] Could not check origin for Phoenix.Socket transport.

Origin of the request: http://localhost:3000

This happens when you are attempting a socket connection to
a different host than the one configured in your config/
files. For example, in development the host is configured
to "localhost" but you may be trying to access it from
"127.0.0.1". To fix this issue, you may either:

  1. update [url: [host: ...]] to your actual host in the
     config file for your current environment (recommended)

  2. pass the :check_origin option when configuring your
     endpoint or when configuring the transport in your
     UserSocket module, explicitly outlining which origins
     are allowed:

        check_origin: ["https://example.com",
                       "//another.com:888", "//other.com"]
```

**Comment by chrismccord** [comment]
A very common cause of issues for folks is the fact that check_origin is true by default in prod, but they fail to properly configure their application for it, so logging the error is definitely what we want to do by default, and as you mentioned, it only appears if the app is misconfigured or a client is intentionally trying to connect from a bad origin. We would be happy to evaluate a PR that adds an option to disable, but logging by default is definitely the way to go for folks. Thanks!

**Comment by thiamsantos** [comment]
> A very common cause of issues for folks is the fact that check_origin is true by default in prod

Makes sense @chrismccord. 

I mainly find strange this log when I already configured the `check_origin` option with the correct domains or custom function.
Wouldn't be possible to only have the error when  `check_origin` is true (the scenario that we want to provide some guidance on how to fix)? And do not log when we already configured the `check_origin` option?
